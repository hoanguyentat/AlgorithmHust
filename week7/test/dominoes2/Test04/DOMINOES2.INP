36
50 35 22
14 14
17 29
41 21
50 24
29 46
20 30
17 23
13 17
9 19
4 9
37 2
35 33
26 38
1 17
28 43
2 49
7 47
47 44
8 33
20 21
20 48
5 26
31 29
8 7
17 9
27 15
50 23
34 19
41 44
49 48
4 14
17 1
31 13
50 46
1 49
21
18
38
19
49
35
3
34
35
15
7
21
8
25
32
17
24
37
24
15
42
33
9 12 2
5 4
9 5
7 7
7 2
3 7
4 8
8 6
8 4
9 3
7 8
6 8
8 9
3
9
17 1 8
4 15
7
7
17
7
5
10
12
3
56 20 8
18 32
8 35
16 41
49 35
35 26
2 9
4 5
30 20
24 10
14 49
21 19
42 12
27 9
3 48
55 5
40 19
30 1
15 30
47 40
23 43
34
10
11
15
56
37
41
11
78 17 56
63 9
60 74
43 62
12 38
73 35
22 44
27 78
43 17
42 34
17 18
74 16
74 3
19 54
15 74
22 35
70 30
11 26
5
11
51
3
66
32
71
4
51
46
36
70
68
60
19
44
74
19
61
32
59
71
32
28
57
49
46
73
11
18
9
65
25
54
15
75
17
23
33
57
4
76
47
3
56
60
2
63
31
30
50
71
78
62
37
23
73 94 45
39 3
9 29
70 37
66 71
17 25
12 1
44 11
27 46
45 13
7 36
16 63
53 56
71 44
12 31
4 36
5 3
45 71
8 20
34 64
72 26
43 1
56 35
49 60
13 46
33 35
65 61
35 6
23 61
65 51
68 44
25 46
12 37
60 2
31 57
29 49
67 20
56 35
3 51
48 36
40 69
13 44
56 6
47 55
68 66
61 27
25 48
19 8
61 45
37 41
19 32
13 65
41 11
62 1
59 63
67 44
29 69
1 39
22 65
61 18
5 52
67 34
12 69
32 59
60 45
20 45
32 66
9 63
6 63
18 48
24 56
33 6
20 33
22 36
64 3
35 68
29 31
63 11
30 17
37 34
71 4
15 21
30 56
35 25
38 66
63 69
59 46
24 72
7 11
46 6
6 12
47 37
52 62
46 47
8 70
69
43
23
36
73
72
29
72
13
11
49
33
71
40
65
34
45
46
6
3
39
1
41
71
33
25
4
13
58
63
66
51
18
28
70
27
41
31
17
6
30
48
52
50
13
47 51 19
31 6
5 32
43 20
4 40
2 41
16 11
28 34
26 34
22 8
31 31
31 6
9 16
47 14
43 1
1 10
28 40
35 45
37 39
11 41
41 17
30 32
19 45
14 23
28 46
11 40
11 17
11 24
33 10
19 14
17 24
7 43
14 45
28 41
15 21
22 1
46 44
43 35
1 19
35 43
39 11
9 33
47 20
12 29
41 1
46 15
34 38
14 27
6 38
40 4
25 34
29 16
38
30
4
27
35
11
16
31
6
21
18
16
32
11
41
34
4
16
5
77 78 15
18 75
38 2
35 12
15 73
30 1
64 58
10 59
13 30
72 63
20 50
51 32
3 50
46 38
69 43
2 61
75 35
75 68
72 11
14 8
47 2
38 45
51 37
62 65
47 6
75 31
22 51
64 24
19 16
60 62
20 1
38 17
61 3
37 32
46 36
23 57
9 28
76 64
44 63
5 51
59 69
27 10
25 69
23 5
7 70
54 77
51 44
14 46
41 11
71 17
22 77
61 35
63 19
32 68
62 23
72 56
3 40
52 6
34 26
62 46
3 67
6 27
41 53
12 22
51 44
36 48
73 23
26 25
76 5
57 32
59 75
30 9
19 62
50 5
36 23
21 46
38 46
35 68
37 20
42
57
1
43
14
8
61
75
44
52
21
77
20
72
19
15 14 9
6 6
6 3
15 11
5 2
4 1
5 15
2 4
2 14
9 2
14 3
2 6
6 8
4 4
13 2
13
8
10
4
14
9
3
11
9
50 32 28
32 2
8 42
40 39
45 32
32 21
2 32
25 38
20 7
31 24
27 3
26 33
49 37
35 30
37 4
50 9
28 45
9 21
23 10
5 18
11 47
7 38
41 29
20 19
7 29
49 48
10 50
30 49
49 40
48 27
7 13
12 11
38 29
42
11
34
15
16
50
3
27
30
23
22
40
48
38
7
1
19
1
13
5
29
35
41
4
19
27
43
43
18 82 2
6 10
8 8
16 14
4 7
8 1
9 13
9 3
8 9
10 5
1 3
6 5
13 13
3 14
1 15
18 11
7 13
1 15
9 17
13 11
7 1
10 10
5 13
11 12
16 3
14 4
9 15
13 13
2 11
3 4
5 14
2 13
10 17
3 11
14 13
17 9
2 11
15 3
2 13
17 11
16 15
8 3
12 15
13 10
10 17
12 7
10 1
5 18
12 7
4 17
14 8
4 12
14 17
10 1
11 5
17 14
7 17
9 18
3 7
14 7
18 7
11 13
2 18
16 9
2 4
2 17
15 13
8 1
12 11
15 5
11 4
2 14
15 1
7 9
5 14
12 18
2 6
2 17
14 12
17 16
9 12
15 11
10 8
14
6
92 77 51
57 22
9 42
79 40
11 40
62 20
3 50
9 26
83 49
39 49
30 91
15 78
74 91
75 5
73 16
5 67
14 12
89 31
40 1
4 3
35 43
17 21
83 24
14 64
73 91
73 89
79 75
59 67
7 64
8 30
23 92
21 59
39 47
51 19
27 51
89 19
19 89
23 14
13 4
87 73
6 28
80 83
11 58
10 35
49 38
25 91
51 84
58 78
3 32
6 37
47 56
84 37
37 44
65 56
89 32
69 15
54 20
41 31
76 83
91 87
17 16
78 92
5 42
3 45
46 91
51 66
11 12
70 8
48 63
47 67
31 25
20 82
78 73
13 62
77 68
56 30
89 30
71 66
25
69
90
67
56
56
18
89
16
59
71
74
53
26
41
50
4
27
52
80
41
60
18
32
83
12
15
69
28
68
60
17
40
45
81
30
86
36
31
19
8
26
10
23
31
13
86
42
51
78
9
82 28 7
8 14
27 82
63 77
1 72
56 80
40 75
49 16
77 41
36 40
65 57
33 61
28 36
12 30
5 77
6 29
69 72
3 39
25 52
76 9
46 10
55 67
79 37
4 46
25 50
76 33
45 23
44 55
34 27
1
37
22
18
66
44
66
46 23 38
44 31
10 4
18 44
17 7
37 42
23 40
27 20
25 26
8 13
4 15
42 20
8 39
24 45
13 16
31 37
45 34
42 13
32 37
39 4
13 12
9 46
8 41
12 3
35
44
14
39
11
31
3
7
3
23
9
22
27
39
44
7
33
18
22
28
45
32
42
43
32
6
38
1
31
42
28
45
7
7
37
11
21
3
28 69 23
9 13
11 20
15 1
14 9
11 19
22 18
28 20
5 4
28 14
17 1
28 22
22 10
16 3
24 19
16 1
24 4
17 10
20 17
3 16
4 23
19 19
16 13
4 8
7 3
2 9
9 18
14 23
4 4
12 26
7 27
18 16
19 4
8 9
15 28
25 9
16 18
25 23
8 22
3 8
14 18
1 7
13 16
25 7
16 15
15 22
2 1
8 9
7 8
9 11
11 3
16 14
1 20
26 3
7 23
6 22
16 25
1 3
26 10
10 23
5 4
12 9
8 18
14 22
6 28
6 27
24 15
21 18
25 15
13 28
2
10
4
10
11
4
21
25
1
9
1
10
9
11
23
5
21
21
15
9
10
4
11
89 86 18
82 80
57 74
80 29
55 14
41 36
69 33
16 19
44 72
72 61
49 65
13 43
23 81
22 28
87 27
61 43
36 38
38 79
87 74
15 80
86 78
89 72
87 57
51 2
60 58
54 76
66 3
8 65
17 49
58 44
31 65
78 22
50 18
74 4
19 73
78 9
71 26
28 24
70 72
40 62
59 27
71 49
34 11
75 81
48 7
29 11
51 82
23 46
28 41
45 72
26 79
33 82
52 23
80 67
34 84
76 6
60 84
77 29
74 15
34 43
63 4
4 55
77 22
84 89
20 59
44 22
15 77
68 19
48 21
49 29
19 48
87 65
88 33
53 74
89 41
39 21
25 73
58 80
89 38
27 29
44 54
14 53
47 82
21 85
65 86
53 65
34 80
4
57
47
59
11
1
16
75
32
68
12
78
27
79
11
63
54
72
92 29 50
25 65
11 88
78 80
17 65
66 91
26 62
28 63
43 14
27 82
15 28
49 29
67 19
55 9
16 53
5 21
19 27
32 74
49 67
81 83
82 64
43 42
92 14
28 78
89 17
48 90
57 28
31 6
49 17
40 53
46
39
22
71
62
38
58
72
52
2
8
37
55
22
74
13
69
23
55
34
92
42
13
45
8
14
66
11
70
38
23
45
60
1
53
39
50
18
23
65
77
47
52
74
3
16
91
32
46
73
88 71 51
44 9
77 65
28 57
10 42
63 15
82 86
36 22
35 19
76 56
19 32
34 16
74 33
66 45
43 58
60 36
9 72
16 12
16 36
51 40
25 57
51 32
74 48
40 34
13 41
76 24
16 45
68 65
77 50
71 60
27 71
22 73
12 41
13 83
32 21
34 28
67 6
59 31
26 49
60 84
28 71
57 71
78 60
17 34
55 22
52 12
54 23
46 42
24 47
12 13
63 19
46 52
55 26
30 51
75 2
57 36
87 39
11 5
55 12
64 11
80 61
40 70
13 43
42 44
22 33
12 10
20 16
50 72
71 22
61 5
11 84
30 11
16
15
44
59
2
86
50
44
41
62
68
57
65
19
75
9
52
33
14
27
46
66
28
80
18
16
8
67
44
28
52
31
62
80
73
81
14
67
86
30
80
27
23
5
80
23
48
64
67
21
74
35 83 7
7 16
22 25
14 5
18 10
27 6
2 21
4 6
26 7
12 21
16 23
17 28
14 29
8 26
23 26
8 11
7 1
6 3
31 2
32 12
20 22
24 5
24 20
22 15
24 9
21 11
6 17
32 35
31 20
2 14
6 26
27 29
29 9
19 22
27 27
34 31
26 17
33 30
34 19
26 22
8 6
5 5
2 14
6 1
29 26
26 13
29 21
35 34
9 29
11 2
4 12
13 6
30 2
25 16
24 13
13 31
19 10
20 16
29 12
11 35
2 34
35 23
34 27
16 24
30 30
28 17
22 11
24 11
32 23
34 8
9 16
7 35
34 29
25 27
26 2
25 2
31 33
6 30
29 26
31 26
30 29
24 21
35 32
25 24
5
20
4
13
31
7
21
78 11 1
31 73
33 30
4 34
17 30
40 6
60 32
18 27
6 2
19 27
28 54
46 41
41
61 57 38
26 50
50 33
6 38
36 7
21 20
57 25
44 43
43 3
3 6
52 30
22 58
49 31
14 24
51 26
25 50
23 61
38 31
16 27
22 17
23 11
4 18
59 9
59 61
32 19
33 46
19 31
37 50
50 3
32 2
28 57
53 54
7 1
4 38
16 61
30 48
7 33
52 26
14 46
11 20
22 34
18 5
7 15
30 24
28 49
32 10
46 34
8 45
44 40
43 57
49 39
29 28
24 17
21 19
61 34
48 14
52 26
44 33
38
58
44
25
51
57
30
24
60
40
43
25
28
45
14
30
58
16
48
25
16
19
19
61
20
55
37
31
34
17
22
53
29
20
9
21
51
19
11 34 8
9 5
11 9
4 7
6 9
10 6
5 7
8 2
9 1
10 9
7 4
9 6
2 5
3 6
1 7
4 11
11 7
11 10
5 4
2 10
5 4
2 9
8 9
4 5
8 10
11 5
9 3
11 4
4 1
5 8
3 1
4 7
8 2
4 10
8 2
6
4
7
8
5
4
10
11
59 85 57
12 50
46 48
33 28
35 45
28 28
58 26
19 31
3 34
34 32
31 10
34 2
30 24
27 55
48 38
33 15
53 47
9 40
26 44
45 21
42 47
6 19
19 57
32 38
22 25
21 8
36 49
5 39
50 30
33 19
48 42
38 6
10 8
55 45
7 12
7 16
45 57
2 47
53 41
5 20
19 56
44 16
6 10
26 47
21 57
59 53
47 51
57 11
42 56
46 2
17 57
5 11
58 25
32 38
29 41
57 24
37 51
17 20
39 34
43 55
6 1
58 34
15 51
44 7
37 39
29 22
7 48
45 52
40 14
30 57
31 24
26 49
54 40
25 41
29 2
4 18
27 58
49 36
29 14
20 42
27 33
35 23
16 7
17 20
30 22
31 9
34
6
30
17
19
40
1
35
1
31
36
41
54
25
48
10
24
10
37
25
32
14
12
25
10
15
39
47
11
25
34
9
28
9
41
36
29
10
59
27
4
44
24
24
29
48
34
22
19
30
20
31
49
31
58
32
50
7 74 4
6 2
7 3
3 1
5 6
6 6
2 3
1 3
7 2
7 2
2 1
2 3
2 6
5 7
3 7
1 3
7 4
7 2
3 4
4 3
4 3
6 1
3 4
1 6
1 5
1 4
7 1
4 1
1 7
3 5
6 5
1 3
5 4
3 6
4 6
3 4
5 5
5 7
1 2
5 2
5 2
5 2
7 1
4 1
4 4
4 1
1 2
4 6
5 4
7 4
2 1
2 3
2 4
3 4
4 5
3 6
5 7
2 7
3 3
5 7
1 4
4 5
7 1
5 5
3 5
1 5
6 3
6 3
2 7
3 6
7 7
6 7
1 5
3 6
6 5
1
6
2
5
15 75 15
9 14
14 2
8 14
15 13
15 2
7 13
15 9
9 3
7 3
9 11
8 14
10 6
6 4
5 8
5 4
8 1
13 15
15 1
5 11
5 2
8 9
9 2
2 7
4 8
10 10
13 7
7 9
2 6
9 6
13 15
9 4
5 8
1 1
13 15
9 2
7 2
9 4
13 12
5 8
8 5
10 1
4 9
2 2
4 4
4 2
13 15
15 5
4 8
15 14
7 11
3 11
12 6
8 2
8 10
8 15
7 1
15 14
8 5
5 12
2 4
4 11
11 2
6 1
1 1
12 4
12 3
14 11
3 4
2 10
9 8
14 14
13 8
1 6
11 10
7 13
15
2
15
6
12
7
13
5
10
1
1
13
4
9
2
96 18 15
74 72
12 52
15 22
43 61
37 38
37 17
65 11
50 27
53 25
66 87
25 61
35 77
12 12
69 62
86 57
92 92
80 43
6 31
6
46
46
21
59
9
80
89
46
48
62
40
92
58
2
56 29 6
53 2
44 9
15 17
1 19
31 35
33 12
46 40
25 25
48 25
54 27
35 10
1 42
42 43
9 53
4 43
18 45
9 6
44 27
20 42
55 7
55 4
14 11
44 32
33 31
32 40
15 6
37 4
11 33
32 27
15
50
7
49
22
28
31 46 15
28 5
27 12
29 4
26 12
6 12
3 16
2 24
7 2
31 13
2 25
6 14
2 17
1 12
4 11
26 23
7 2
2 21
28 20
18 22
29 27
19 3
28 27
6 18
13 11
3 11
28 7
9 10
17 8
21 19
26 15
31 16
16 4
24 26
24 22
18 9
29 20
1 31
17 2
17 15
27 10
4 15
10 16
14 18
27 2
1 14
13 5
8
10
12
11
4
22
28
5
11
27
8
25
14
7
5
70 70 38
9 54
62 48
20 61
67 19
23 28
43 27
3 62
7 65
66 39
45 34
14 8
47 55
5 6
5 3
16 37
49 12
42 30
65 42
10 67
16 25
43 24
50 16
48 4
40 28
22 2
35 39
45 61
38 50
41 42
70 15
55 16
6 28
69 68
52 59
45 63
45 35
47 47
65 42
31 19
21 15
24 66
31 3
7 45
39 40
24 44
39 46
29 34
23 38
41 66
41 64
25 41
50 18
6 44
5 4
37 12
69 28
57 19
19 9
14 31
31 58
11 23
3 66
45 18
10 27
41 67
26 47
29 59
5 50
69 37
8 59
41
49
4
17
11
46
47
8
26
64
14
59
40
7
26
28
26
24
23
46
46
7
23
45
21
47
38
64
16
44
68
44
6
68
15
3
32
66
20 60 15
19 5
12 20
13 10
3 17
17 16
7 9
17 1
14 11
8 10
14 18
4 13
18 10
14 14
10 15
1 2
15 18
14 4
12 10
18 12
9 9
4 15
6 15
12 4
2 14
16 20
20 3
6 1
9 12
2 10
20 19
2 12
11 17
3 12
14 18
2 15
4 8
19 10
16 5
4 8
10 15
1 12
15 14
15 15
7 5
7 13
4 13
16 7
9 4
19 13
17 19
18 5
18 1
14 20
5 11
14 19
10 4
5 9
15 19
7 15
10 8
11
12
8
16
15
17
16
13
13
4
20
9
13
17
16
12 58 5
5 12
1 11
6 1
3 7
8 6
7 5
11 10
8 1
10 6
8 8
11 5
9 11
4 1
7 11
5 10
2 9
12 9
5 6
5 3
12 10
5 5
10 9
5 6
2 6
7 7
10 7
5 7
8 8
1 3
4 3
8 11
7 8
5 1
7 2
4 1
3 2
12 3
8 3
10 5
7 5
3 4
9 2
3 12
1 9
3 10
5 5
4 6
12 6
9 1
6 2
8 5
4 11
7 6
5 3
4 7
1 8
9 9
10 8
10
11
8
12
1
31 13 29
7 21
12 11
9 24
8 1
27 20
1 16
24 28
28 17
27 27
5 27
5 1
17 24
12 23
10
10
24
26
13
20
22
1
9
5
31
23
27
8
28
24
4
31
4
29
30
9
4
19
3
4
30
24
17
54 21 25
54 20
15 17
7 29
36 15
12 5
53 20
16 10
16 39
43 37
31 28
30 13
54 15
51 40
12 54
41 11
10 41
10 46
4 35
24 52
18 11
10 33
16
28
33
46
16
17
51
16
2
36
30
45
32
32
6
13
32
37
39
14
24
46
24
25
26
74 7 70
16 61
16 65
64 20
34 74
30 49
16 9
1 50
34
52
31
69
52
43
43
32
8
55
53
59
19
27
68
54
8
68
45
55
19
55
1
11
1
48
70
65
25
72
47
1
40
30
48
58
60
18
4
36
35
33
20
57
30
14
22
69
52
73
30
58
32
65
1
66
1
44
34
30
2
53
6
69
46
1
49
43
26
4
46 93 18
36 44
17 6
30 8
42 37
39 3
38 7
45 4
14 10
29 42
26 22
16 22
6 6
33 40
35 24
15 20
4 23
25 31
32 34
23 45
20 22
4 18
43 29
1 26
46 31
3 21
44 26
19 18
7 18
10 5
30 13
25 35
41 24
25 37
10 4
14 39
24 26
42 2
19 35
18 46
42 27
7 28
15 26
27 31
8 32
17 19
7 40
1 42
17 8
18 8
12 24
1 24
31 35
29 31
15 45
27 29
1 34
42 24
2 17
13 23
4 4
13 39
8 33
21 2
37 1
36 39
12 7
39 25
11 15
40 5
33 4
14 25
25 25
14 26
14 42
37 8
36 33
36 38
3 1
22 4
19 25
31 43
38 4
7 16
40 21
41 41
37 43
34 42
2 9
17 39
46 9
26 36
15 34
15 13
8
29
25
23
44
4
37
37
23
9
19
33
37
38
2
9
24
20
25 31 5
17 23
13 8
1 16
4 15
12 24
15 15
17 7
5 8
22 3
16 19
6 4
17 16
9 13
2 21
20 3
9 19
20 22
5 11
23 3
13 10
10 5
15 19
2 3
21 19
19 14
17 7
19 5
17 21
5 3
13 25
10 22
1
20
24
25
14
