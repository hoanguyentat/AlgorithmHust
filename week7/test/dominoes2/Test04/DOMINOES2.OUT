27
8
6
10
46
53
26
47
11
32
17
67
9
33
24
48
40
58
34
1
39
11
55
7
15
20
8
25
51
19
12
21
25
47
38
12